rpi3-convert-stage-5.sh

```diff
@@ -16,26 +16,60 @@
 
 application_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 output_dir=${application_dir}/output
+uboot_dir=${output_dir}/uboot-mender
 bin_dir_pi=${output_dir}/bin/raspberrypi
 sdimg_base_dir=$output_dir/sdimg
 
 echo "Running: $(basename $0)"
 declare -a sdimgmappings
 declare -a sdimg_partitions=("boot" "primary" "secondary" "data")
 
+# Takes following arguments:
+#
+#  $1 - ARM toolchain
 build_uboot_files() {
+  local CROSS_COMPILE=${1}-
+  local ARCH=arm
+  local revision="mender-rpi-2017.09"
+  local uboot_repo_vc_dir=$uboot_dir/.git
+  local repo_clean=0
+
+  export CROSS_COMPILE=$CROSS_COMPILE
+  export ARCH=$ARCH
+
   mkdir -p $bin_dir_pi
 
-  echo -e "Downloading U-Boot related files..."
+  echo -e "Building U-Boot related files..."
+
+  if [ ! -d $uboot_repo_vc_dir ]; then
+    git clone https://github.com/mendersoftware/uboot-mender.git -b $revision
+    repo_clean=1
+  fi
+
+  cd $uboot_dir
+
+  [[ repo_clean -eq 0 ]] && { make --quiet clean; make --quiet distclean; }
```

Comment:
Seems a bit unnecessary to have a conditional here. 

You can always run the the following sequence. 

```
make --quiet distclean
make rpi_3_32b_defconfig
make
make envtools
```

`distclean`will succeed even if there is nothing to clean. 

You can drop `make clean`, `distclean` target has a dependency on `clean` target.

Reply:
OK